### PR TITLE
[SELC-5507] feat: namirial implementation in onboarding-crypto sdk

### PR DESCRIPTION
**Pull request by giulia-tremolada**
#### List of Changes
- add namirial implementation
- upgrade sdk version from 0.2.1 to 0.2.2

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: libs/onboarding-sdk-crypto/pom.xml [hunk 1]
```diff
@@ -75,6 +75,25 @@
             <version>2.1.3</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.http-client</groupId>
+            <artifactId>google-http-client</artifactId>
+            <version>1.40.1</version> <!-- Check for the latest version -->
```
Comment (manuraf) [inline libs/onboarding-sdk-crypto/pom.xml:81]: you could use a property <google-http-sdk-version> for version in order to use also in the below dependencies

File: libs/onboarding-sdk-crypto/pom.xml [hunk 2]
```diff
@@ -75,6 +75,25 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.google.http-client</groupId>
+            <artifactId>google-http-client-apache-v2</artifactId> <!-- If you prefer to use Apache HTTP transport -->
+            <version>1.40.1</version>
```
Comment (manuraf) [inline libs/onboarding-sdk-crypto/pom.xml:86]: ditto

File: libs/onboarding-sdk-crypto/pom.xml [hunk 3]
```diff
@@ -75,6 +75,25 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.google.http-client</groupId>
+            <artifactId>google-http-client-jackson2</artifactId>
+            <version>1.41.5</version> <!-- You can use the latest version if needed -->
```
Comment (manuraf) [inline libs/onboarding-sdk-crypto/pom.xml:91]: ditto

File: libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/Pkcs7HashSignService.java [hunk 4]
```diff
@@ -3,4 +3,6 @@
 import org.apache.pdfbox.pdmodel.interactive.digitalsignature.SignatureInterface;
 
 public interface Pkcs7HashSignService extends SignatureInterface {
+
+    boolean returnsFullPdf();
```
Comment (manuraf) [inline libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/Pkcs7HashSignService.java:7]: you could add some comments about behavior of this  method

File: libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/PadesSignServiceImpl.java [hunk 5]
```diff
@@ -20,21 +22,27 @@ public PadesSignServiceImpl(Pkcs7HashSignService pkcs7Signature) {
...
-            signature.setLocation(signInfo.getLocation());
-            signature.setReason(signInfo.getReason());
-            signature.setSignDate(Calendar.getInstance());
-            SignatureOptions signatureOptions = new SignatureOptions();
-            signatureOptions.setPreferredSignatureSize(35944);
-            doc.addSignature(signature, this.pkcs7Signature, signatureOptions);
-            doc.saveIncremental(fos);
-
+        try (FileOutputStream fos = new FileOutputStream(signedPdfFile)){
+            if(this.pkcs7Signature.returnsFullPdf()){
```
Comment (manuraf) [inline libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/PadesSignServiceImpl.java:26]: ditto comments

**Review by manuraf — CHANGES_REQUESTED** [review]

File: libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/client/NamirialHttpClient.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        ObjectMapper objectMapper = new ObjectMapper();
+
+
+        String credentials = objectMapper.writeValueAsString(request.getCredentials());
+        String preferences = objectMapper.writeValueAsString(request.getPreferences());
+
+
+        multipartContent.addPart(
+                new MultipartContent.Part(
+                        new HttpHeaders().set("Content-Disposition", "form-data; name=\"file\"; filename=\""+ request.getFile().getAbsolutePath() +"\""),
```
Comment (gunzip) [inline libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/client/NamirialHttpClient.java:34]: Do we need to send the whole file path to Namirial?

```suggestion
                        new HttpHeaders().set("Content-Disposition", "form-data; name=\"file\"; filename=\""+ request.getFile().getName() +"\""),
```

Comment (giulia-tremolada) [reply libs/onboarding-sdk-crypto/src/main/java/it/pagopa/selfcare/onboarding/crypto/client/NamirialHttpClient.java]: fixed it

**Review by manuraf — APPROVED** [review]

**Pull request merged**